packages/account/src/SmartAccount.ts

```diff
@@ -140,10 +139,7 @@ export abstract class SmartAccount implements ISmartAccount {
           finalUserOp.preVerificationGas = preVerificationGas ?? userOp.preVerificationGas;
           finalUserOp.paymasterAndData = paymasterAndData ?? userOp.paymasterAndData;
         } else {
-          // use dummy values for gas limits as fee quote call will ignore this later.
-          finalUserOp.callGasLimit = DefaultGasLimit.callGasLimit;
-          finalUserOp.verificationGasLimit = DefaultGasLimit.verificationGasLimit;
-          finalUserOp.preVerificationGas = DefaultGasLimit.preVerificationGas;
+          throw new Error("Either pass skipBundlerGasEstimation = false OR pass paymasterServiceData with mode as SPONSORED");
```

Comment:
This is the case when they don't send the `PaymasterMode.SPONSORED` and needed in token paymaster.

There is another else case below with 
```js
else {
        Logger.warn("Skipped paymaster call. If you are using paymasterAndData, generate data externally");
        finalUserOp = await this.calculateUserOpGasValues(userOp);
        finalUserOp.paymasterAndData = "0x";
      }
```

Reply:
Something like this
https://www.diffchecker.com/n6sKSvjd/